Gemfile

```diff
@@ -18,6 +18,10 @@ rack_version = nil if rack_version.empty? || (rack_version == 'stable')
 rack_version = { github: 'rack/rack' } if rack_version == 'master'
 gem 'rack', rack_version
 
+if rack_version == 'master'
```

Comment:
The `rack_version` local variable gets overwritten in line 18 so it will never be `master` here, it will be either `nil` or `{ github: 'rack/rack' }` Maybe it could be rewritten to something like this:

```ruby
rack_version = ENV.fetch('rack', nil)
if rack_version == 'master'
  gem 'rack', { github: 'rack/rack' }
  gem 'rack-session'
end
```

Reply:
Ohhhhhhh I see...